frontend/src/features/codes/OwnersList.tsx

```diff
@@ -0,0 +1,113 @@
+import {IconCrossCircle, IconSearch, TextInput} from "hds-react";
+import {useCallback, useRef, useState} from "react";
+import {useGetOwnersQuery} from "../../app/services";
+import {ModifyOwnerModal, QueryStateHandler} from "../../common/components";
+import {IFilterOwnersQuery, IOwner} from "../../common/schemas";
+
+const OwnerResultList: React.FC<{params: IFilterOwnersQuery}> = ({params}) => {
+    const MIN_LENGTH = 2; // Minimum length before applying filter
+    const MAX_ROWS = 12; // Maximum rows to show
+    if (params.name?.length && params.name?.length < MIN_LENGTH) {
+        delete params.name;
+    }
+    if (params.identifier?.length && params.identifier?.length < MIN_LENGTH) {
+        delete params.identifier;
+    }
+    // get the data
+    const {data, error, isLoading} = useGetOwnersQuery({...params, limit: MAX_ROWS});
+
+    // state for the modal
+    const [isModifyOwnerModalOpen, setIsModifyOwnerModalOpen] = useState(false);
+    const [owner, setOwner] = useState<IOwner | undefined>(undefined);
+
+    // action for the row click
+    const editFn = (ownerEdited: IOwner) => {
+        setIsModifyOwnerModalOpen(true);
+        setOwner(ownerEdited);
+    };
+    return (
+        <QueryStateHandler
+            data={data}
+            error={error}
+            isLoading={isLoading}
+        >
+            <div className="list-headers">
+                <div>Nimi</div>
+                <div>Henkilö- tai Y-tunnus</div>
+            </div>
+            <ul className="results-list">
+                {data?.contents.map((owner: IOwner) => (
+                    <div
+                        key={owner.identifier}
```

Comment:
`owner.identifier` is a bad value for the `key`, as it may be duplicate or `null`. If you have owners without an identifier this listing will break.

Reply:
Good point. I changed the key to use index instead.

Reply:
Index is certainly better than `owner.identifier`, but `owner.id` would be even better 😉 

Using Index as the key is react's way of doing things if key is not provided at all, but it has some issues as well https://react.dev/learn/rendering-lists#why-does-react-need-keys

Reply:
Changed it once more...